scripts/update-dependencies.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const { isEqual, cloneDeep } = require('lodash');
+const {
+  listAllPackages,
+  runInDir,
+  updatePackageJson,
+  findMonorepoRoot,
+  withProgress,
+} = require('@mongodb-js/monorepo-tools');
+
+const UPDATE_CONFIGS = require('./update-dependencies-config');
+
+async function hoistSharedDependencies(newVersions) {
```

Comment:
To be very clear, this flow is definitely sub optimal, we run install three time, but it was also the only way we could consistently make npm do what we want instead of whatever it decides to do wrt shared dependencies and their current location in the dependency tree and debugging npm behavior on a pretty complex dependency tree is really really hard and I'm not sure it's worth it

Reply:
When we decide to update other deps using this scripts, they will also be hoisted at the top. And then running locally npm bootstrap, _would it_ create updates in lock file?

Reply:
I tested it with a few different dependencies and lock stays stable (meaning nothing is unexpectedly unhoisted) after running script a couple of times with different deps, but to be honest I don't know how to validate that this will never happen at all, I lost any trust in npm internal algorithms at this point 😆